src/core/main.c

```diff
@@ -1977,13 +1975,13 @@ static int invoke_main_loop(
         assert(ret_error_message);
 
         for (;;) {
-                r = manager_loop(m);
-                if (r < 0) {
+                ManagerObjective obj = manager_loop(m);
+                if (obj < 0) {
                         *ret_error_message = "Failed to run main loop";
-                        return log_emergency_errno(r, "Failed to run main loop: %m");
+                        return log_emergency_errno(obj, "Failed to run main loop: %m");
```

Comment:
This is out of the scope in this commit, but is it not necessary to set `ret_retval` here?

Reply:
With the changes done in patch 1/2, if we return an error, we don't try to shut down, so those parameters are never used. In other words, we follow the usual pattern of "set output variables only on success".